item.rb

```diff
@@ -0,0 +1,28 @@
+require 'securerandom'
+require 'date'
+
+class Item
+  attr_accessor :label, :genre, :author
+  attr_reader :id, :published_date, :archived
+
+  def initialize(published_date = nil, id: nil)
+    @id = id || SecureRandom.uuid
+    @label = nil
+    @genre = nil
+    @author = nil
+    @published_date = published_date
+    @archived = false
+    move_to_archive
+  end
+
+  def can_be_archived?
+    return false if @published_date.nil?
+
+    published_date = DateTime.parse(@published_date)
+    published_date < DateTime.now - 30
+  end
+
+  def move_to_archive
+    @archived = can_be_archived?
+  end
```

Comment:
- [ ] As required for the project the Item class to have a `move_to_archive()` method which  reuses `can_be_archived?`method, changing the archived property to true if the result of the `can_be_archived?()` method is true, and do nothing if the result of the `can_be_archived?()` method is false, kindly ensure that you also update this to match the expected result.
```ruby
# example
def move_to_archive
      if can_be_archived?
        @archived = true
      end
    end
```